<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>KFiltre</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="dockNestingEnabled">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <property name="margin">
     <number>3</number>
    </property>
    <property name="horizontalSpacing">
     <number>6</number>
    </property>
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="sizeConstraint">
       <enum>QLayout::SetMaximumSize</enum>
      </property>
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex">
         <number>-1</number>
        </property>
        <property name="tabsClosable">
         <bool>false</bool>
        </property>
        <property name="movable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_as"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
   </widget>
   <widget class="QMenu" name="menuImage">
    <property name="title">
     <string>Image</string>
    </property>
    <addaction name="actionCrop"/>
    <addaction name="actionGrayscale"/>
    <addaction name="actionFusion"/>
    <addaction name="actionResize"/>
    <addaction name="separator"/>
    <addaction name="actionVertical_Resize"/>
    <addaction name="actionHorizontal_Resize"/>
    <addaction name="separator"/>
    <addaction name="actionRotate_90"/>
    <addaction name="actionRotate_270"/>
    <addaction name="actionVertical_Mirror"/>
    <addaction name="actionHorizontal_Mirror"/>
   </widget>
   <widget class="QMenu" name="menuHistogram">
    <property name="title">
     <string>Histogram</string>
    </property>
    <addaction name="actionRGB"/>
    <addaction name="actionYUV"/>
    <addaction name="actionEqualize"/>
    <addaction name="actionNormalize"/>
   </widget>
   <widget class="QMenu" name="menuFilters">
    <property name="title">
     <string>Filters</string>
    </property>
    <addaction name="actionBlur"/>
    <addaction name="actionNegative"/>
    <addaction name="actionPaint"/>
    <addaction name="actionSmooth"/>
    <addaction name="separator"/>
    <addaction name="actionEdge_Detection"/>
    <addaction name="actionEdge_Enhancement"/>
    <addaction name="separator"/>
    <addaction name="actionCustom"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>?</string>
    </property>
    <addaction name="actionAbout_us"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <widget class="QMenu" name="menuSelect">
     <property name="title">
      <string>Select</string>
     </property>
     <addaction name="actionSelect_all"/>
     <addaction name="actionNone"/>
     <addaction name="actionRectangle"/>
     <addaction name="actionEllipse"/>
     <addaction name="actionPath"/>
    </widget>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="menuSelect"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuImage"/>
   <addaction name="menuHistogram"/>
   <addaction name="menuFilters"/>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="cursor">
    <cursorShape>ArrowCursor</cursorShape>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="allowedAreas">
    <set>Qt::AllToolBarAreas</set>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
   <addaction name="separator"/>
   <addaction name="actionSelect_all"/>
   <addaction name="actionNone"/>
   <addaction name="actionRectangle"/>
   <addaction name="actionEllipse"/>
   <addaction name="actionPath"/>
   <addaction name="actionCrop"/>
   <addaction name="separator"/>
   <addaction name="actionRotate_270"/>
   <addaction name="actionRotate_90"/>
   <addaction name="actionHorizontal_Mirror"/>
   <addaction name="actionVertical_Mirror"/>
  </widget>
  <action name="actionOpen">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>icons/document-open.png</normaloff>icons/document-open.png</iconset>
   </property>
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="iconText">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
   <property name="visible">
    <bool>true</bool>
   </property>
   <property name="menuRole">
    <enum>QAction::TextHeuristicRole</enum>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset>
     <normaloff>icons/document-save.png</normaloff>icons/document-save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
   <property name="toolTip">
    <string>Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionCrop">
   <property name="icon">
    <iconset>
     <normaloff>icons/transform-crop.png</normaloff>icons/transform-crop.png</iconset>
   </property>
   <property name="text">
    <string>Crop</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionGrayscale">
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>icons/layer-visible-off.png</normaloff>icons/layer-visible-off.png</iconset>
   </property>
   <property name="text">
    <string>Grayscale</string>
   </property>
  </action>
  <action name="actionFusion">
   <property name="icon">
    <iconset>
     <normaloff>icons/insert-image.png</normaloff>icons/insert-image.png</iconset>
   </property>
   <property name="text">
    <string>Fusion...</string>
   </property>
  </action>
  <action name="actionResize">
   <property name="icon">
    <iconset>
     <normaloff>icons/transform-scale.png</normaloff>icons/transform-scale.png</iconset>
   </property>
   <property name="text">
    <string>Resize...</string>
   </property>
  </action>
  <action name="actionRGB">
   <property name="icon">
    <iconset>
     <normaloff>icons/preferences-activities.png</normaloff>icons/preferences-activities.png</iconset>
   </property>
   <property name="text">
    <string>RGB</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="actionNormalize">
   <property name="text">
    <string>Normalize</string>
   </property>
  </action>
  <action name="actionCustom">
   <property name="text">
    <string>Custom...</string>
   </property>
  </action>
  <action name="actionAbout_us">
   <property name="icon">
    <iconset>
     <normaloff>icons/help-about.png</normaloff>icons/help-about.png</iconset>
   </property>
   <property name="text">
    <string>About us</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="icon">
    <iconset>
     <normaloff>icons/document-save-as.png</normaloff>icons/document-save-as.png</iconset>
   </property>
   <property name="text">
    <string>Save as...</string>
   </property>
   <property name="toolTip">
    <string>Save as</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionYUV">
   <property name="icon">
    <iconset>
     <normaloff>icons/view-object-histogram-logarithmic.png</normaloff>icons/view-object-histogram-logarithmic.png</iconset>
   </property>
   <property name="text">
    <string>YUV</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+H</string>
   </property>
  </action>
  <action name="actionSelect_all">
   <property name="icon">
    <iconset>
     <normaloff>icons/select_all.png</normaloff>icons/select_all.png</iconset>
   </property>
   <property name="text">
    <string>All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionNone">
   <property name="icon">
    <iconset>
     <normaloff>icons/dialog-cancel.png</normaloff>icons/dialog-cancel.png</iconset>
   </property>
   <property name="text">
    <string>None</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+A</string>
   </property>
  </action>
  <action name="actionRectangle">
   <property name="icon">
    <iconset>
     <normaloff>icons/draw-rectangle.png</normaloff>icons/draw-rectangle.png</iconset>
   </property>
   <property name="text">
    <string>Rectangle</string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset>
     <normaloff>icons/edit-undo.png</normaloff>icons/edit-undo.png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset>
     <normaloff>icons/edit-redo.png</normaloff>icons/edit-redo.png</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="actionBlur">
   <property name="text">
    <string>Blur</string>
   </property>
  </action>
  <action name="actionEdge_Enhancement">
   <property name="text">
    <string>Edge Enhancement</string>
   </property>
  </action>
  <action name="actionPaint">
   <property name="text">
    <string>Paint</string>
   </property>
  </action>
  <action name="actionEdge_Detection">
   <property name="text">
    <string>Edge Detection</string>
   </property>
  </action>
  <action name="actionVertical_Resize">
   <property name="text">
    <string>Smart Vertical Resize</string>
   </property>
   <property name="shortcut">
    <string>V</string>
   </property>
  </action>
  <action name="actionHorizontal_Resize">
   <property name="text">
    <string>Smart Horizontal Resize</string>
   </property>
   <property name="shortcut">
    <string>H</string>
   </property>
  </action>
  <action name="actionNegative">
   <property name="text">
    <string>Negative</string>
   </property>
  </action>
  <action name="actionSmooth">
   <property name="text">
    <string>Smooth</string>
   </property>
  </action>
  <action name="actionEqualize">
   <property name="text">
    <string>Equalize</string>
   </property>
  </action>
  <action name="actionRotate_90">
   <property name="icon">
    <iconset>
     <normaloff>icons/object-rotate-right.png</normaloff>icons/object-rotate-right.png</iconset>
   </property>
   <property name="text">
    <string>Rotate 90°</string>
   </property>
  </action>
  <action name="actionRotate_270">
   <property name="icon">
    <iconset>
     <normaloff>icons/object-rotate-left.png</normaloff>icons/object-rotate-left.png</iconset>
   </property>
   <property name="text">
    <string>Rotate 270°</string>
   </property>
  </action>
  <action name="actionVertical_Mirror">
   <property name="icon">
    <iconset>
     <normaloff>icons/object-flip-horizontal.png</normaloff>icons/object-flip-horizontal.png</iconset>
   </property>
   <property name="text">
    <string>Vertical Mirror</string>
   </property>
  </action>
  <action name="actionHorizontal_Mirror">
   <property name="icon">
    <iconset>
     <normaloff>icons/object-flip-vertical.png</normaloff>icons/object-flip-vertical.png</iconset>
   </property>
   <property name="text">
    <string>Horizontal Mirror</string>
   </property>
  </action>
  <action name="actionEllipse">
   <property name="icon">
    <iconset>
     <normaloff>icons/draw-ellipse.png</normaloff>icons/draw-ellipse.png</iconset>
   </property>
   <property name="text">
    <string>Ellipse</string>
   </property>
   <property name="shortcut">
    <string>E</string>
   </property>
  </action>
  <action name="actionPath">
   <property name="icon">
    <iconset>
     <normaloff>icons/draw-polyline.png</normaloff>icons/draw-polyline.png</iconset>
   </property>
   <property name="text">
    <string>Path</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionClose</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_as</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionGrayscale</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>toGray()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRGB</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showHistogram()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionYUV</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showYUVHistogram()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCrop</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>crop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRectangle</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setRectTool()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSelect_all</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>selectAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNone</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>cancelSelection()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionUndo</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRedo</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionBlur</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>applyBlur()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEdge_Enhancement</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>applyEdgeEnhancement()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPaint</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>applyPaintEffect()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEdge_Detection</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>applyEdgeDetection()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFusion</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fusion()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionResize</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showResizeDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNormalize</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>normalizeHistogram()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionVertical_Resize</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>smartResizeV()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionHorizontal_Resize</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>smartResizeH()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSmooth</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>smooth()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNegative</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>negative()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCustom</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showCustomDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEqualize</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>equalizeHistogram()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRotate_90</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>rotateClockwise()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRotate_270</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>rotateCounterClockwise()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionVertical_Mirror</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>verticalMirror()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionHorizontal_Mirror</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>horizontalMirror()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEllipse</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setEllipseTool()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPath</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setPathTool()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>open()</slot>
  <slot>save()</slot>
  <slot>saveAs()</slot>
  <slot>showHistogram()</slot>
  <slot>showYUVHistogram()</slot>
  <slot>toGray()</slot>
  <slot>crop()</slot>
  <slot>setRectTool()</slot>
  <slot>selectAll()</slot>
  <slot>cancelSelection()</slot>
  <slot>moveSelection()</slot>
  <slot>undo()</slot>
  <slot>redo()</slot>
  <slot>applyBlur()</slot>
  <slot>applyEdgeEnhancement()</slot>
  <slot>applyPaintEffect()</slot>
  <slot>applyEdgeDetection()</slot>
  <slot>fusion()</slot>
  <slot>showResizeDialog()</slot>
  <slot>normalizeHistogram()</slot>
  <slot>smartResizeV()</slot>
  <slot>smartResizeH()</slot>
  <slot>smooth()</slot>
  <slot>negative()</slot>
  <slot>showCustomDialog()</slot>
  <slot>equalizeHistogram()</slot>
  <slot>rotateClockwise()</slot>
  <slot>rotateCounterClockwise()</slot>
  <slot>verticalMirror()</slot>
  <slot>horizontalMirror()</slot>
  <slot>setEllipseTool()</slot>
  <slot>setPathTool()</slot>
 </slots>
</ui>
